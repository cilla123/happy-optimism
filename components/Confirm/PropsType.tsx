export default interface PropsType {
  shape?: 'radius';
  visible?: boolean;
  animationType?:
    'fade' | 'door' | 'flip' | 'rotate' | 'zoom' |
    'moveUp' | 'moveDown' | 'moveLeft' | 'moveRight' |
    'slideUp' | 'slideDown' | 'slideLeft' | 'slideRight';
  animationDuration?: number;
  width?: string | number;
  onMaskClick?: () => void;
  title?: any;
  message?: any;
  okText?: string;
  onOk?: () => void;
  cancelText?: string;
  onCancel?: () => void;
}

declare global {
  interface Window {
    zarmConfirm?: any;
  }
}
