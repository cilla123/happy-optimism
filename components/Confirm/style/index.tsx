import '../../style/';
import '../../Modal/style/';
import './index.scss';
