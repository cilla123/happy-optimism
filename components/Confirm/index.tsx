import React, { PureComponent } from 'react';
// import ReactDOM from 'react-dom';
import classnames from 'classnames';
import PropsType from './PropsType';
import Modal from '../Modal';

export interface ConfirmProps extends PropsType {
  prefixCls?: string;
  className?: string;
}

export default class Confirm extends PureComponent<ConfirmProps, {}> {

  static defaultProps = {
    prefixCls: 'za-confirm',
    animationType: 'zoom',
    message: '',
    okText: '确定',
    cancelText: '取消',
  };

  // static show = (props) => {
  //   ReactDOM.render(<Confirm {...props} visible />, window.zarmConfirm);
  // }

  // static hide = () => {
  //   ReactDOM.render(<Confirm visible={false} />, window.zarmConfirm);
  // }

  render() {
    const { prefixCls, className, title, message, okText, cancelText, onOk, onCancel, ...others } = this.props;
    const cls = classnames(prefixCls, className);

    return (
      <Modal className={cls} {...others}>
        <Modal.Header title={title} />
        <Modal.Body className={`${prefixCls}-body`}>
          {message}
        </Modal.Body>
        <Modal.Footer className={`${prefixCls}-footer`}>
          <a className={`${prefixCls}-footer-button block`} onClick={onCancel}>{cancelText}</a>
          <a className={`${prefixCls}-footer-button block ${prefixCls}-footer-button-ok`} onClick={onOk}>{okText}</a>
        </Modal.Footer>
      </Modal>
    );
  }
}

// if (!window.zarmConfirm) {
//   window.zarmConfirm = document.createElement('div');
//   document.body.appendChild(window.zarmConfirm);
// }

// ReactDOM.render(<Confirm visible={false} />, window.zarmConfirm);
