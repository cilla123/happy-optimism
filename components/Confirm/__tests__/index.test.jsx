import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Confirm from '../index';

describe('Confirm', () => {
  it('renders correctly', () => {
    const onOk = jest.fn();
    const onCancel = jest.fn();

    const wrapper = render(
      <Confirm
        radius
        visible
        title="确认信息"
        message="你确定要这样做吗？"
        onOk={onOk}
        onCancel={onCancel}
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
