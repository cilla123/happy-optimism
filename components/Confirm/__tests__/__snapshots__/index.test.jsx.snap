// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm renders correctly 1`] = `
<div
  class="za-modal za-confirm"
  style="-webkit-animation-duration:200ms;animation-duration:200ms"
>
  <div
    class="za-modal-wrapper"
  >
    <div
      class="za-modal-dialog"
      style="width:70%;-webkit-animation-duration:200ms;animation-duration:200ms"
    >
      <div
        class="za-modal-header"
      >
        <div
          class="za-modal-header-title"
        >
          确认信息
        </div>
      </div>
      <div
        class="za-modal-body za-confirm-body"
      >
        你确定要这样做吗？
      </div>
      <div
        class="za-modal-footer za-confirm-footer"
      >
        <a
          class="za-confirm-footer-button block"
        >
          取消
        </a>
        <a
          class="za-confirm-footer-button block za-confirm-footer-button-ok"
        >
          确定
        </a>
      </div>
    </div>
  </div>
  <div
    class="za-mask normal"
  />
</div>
`;
